design-documents/testing/integration/data-fixtures.md

```diff
@@ -0,0 +1,154 @@
+## Problem
+When it comes to write a data fixture for your test case, the first thing you would do is to search for existing data fixture you can reuse in your test case. Most of the time you will find such data fixture that almost meets the requirements of your test case except that it's missing something very important to your test case.
+Therefore you end up writing a new data fixture for your test case that is 99% a copy of existing one.
+Here is a real example:
+
+- dev/tests/integration/testsuite/Magento/Catalog/_files/product_virtual.php
+- dev/tests/integration/testsuite/Magento/Catalog/_files/product_virtual_in_stock.php
+- dev/tests/integration/testsuite/Magento/Catalog/_files/product_virtual_out_of_stock.php
+
+All these 3 files create a virtual product except one marks the product as out of stock and the other assigns different quantity to the product.
+So if you need a virtual product with available quantity 1, you will probably have to:
+
+- Write a new data fixture (copy-past)
+- Reuse one of these data fixtures in a new data fixture file and edit the quantity there.
+- Reuse one of these data fixtures in your test case and edit the quantity in the test directly.
+ 
+## Solution
+
+We could certainly redesign data fixtures to be object oriented. But if we just want to tackle this issue as simpler as we can, we could consist we could extend the format of data fixture annotation
+to support a second parameter which will be injected to the data fixture file as following.
+```php
+/**
+ * @magentoDataFixture Magento/Catalog/_files/product_virtual.php, {"productData":{"stock_data": {"qty":1}}}
```

Comment:
As I understand, there will be only one argument always - a JSON string, and it may include multiple keys in case the fixture creates multiple different instances? Just wanted to clarify.

Should we not use "," to be consistent with other annotations? Like the one for `@magentoConfigFixture`.